src/API/WP/OAuthService.php

```diff
@@ -91,31 +98,65 @@ public function get_auth_url( string $path ): string {
 					'scope'         => self::SCOPE,
 					'state'         => $state,
 				],
-				self::AUTH_URL
+				$this->get_wpcom_api_url( self::AUTH_URL )
 			)
 		);
 
 		return $auth_url;
 	}
 
 	/**
-	 * Calls an API by Google to get required information in order to form an auth URL.
+	 * Get a WPCOM REST API URl concatenating the endpoint with the API Domain
 	 *
-	 * TODO: Call an actual API by Google.
-	 * We'd probably need use WCS to communicate with the new API.
+	 * @param string $endpoint The endpoint to get the URL for
+	 *
+	 * @return string The WPCOM endpoint with the domain.
+	 */
+	protected function get_wpcom_api_url( string $endpoint ): string {
+		return self::WPCOM_API_URL . $endpoint;
+	}
+
+	/**
+	 * Calls an API by Google via WCS to get required information in order to form an auth URL.
 	 *
 	 * @return array{client_id: string, redirect_uri: string, nonce: string} An associative array contains required information that is retrived from Google.
 	 * client_id:    Google's WPCOM app client ID, will be used to form the authorization URL.
 	 * redirect_uri: A Google's URL that will be redirected to when the merchant approve the app access. Note that it needs to be matched with the Google WPCOM app client settings.
 	 * nonce:        A string returned by Google that we will put it in the auth URL and the redirect_uri. Google will use it to verify the call.
 	 */
 	protected function get_data_from_google(): array {
-		$nonce = 'nonce-123';
+		/** @var Middleware $middleware */
+		$middleware = $this->container->get( Middleware::class );
+		$response   = $middleware->get_sdi_auth_params();
+		$nonce      = $response['nonce'];
 		$this->options->update( OptionsInterface::GOOGLE_WPCOM_AUTH_NONCE, $nonce );
 		return [
-			'client_id'    => '91299',
-			'redirect_uri' => 'https://woo.com',
+			'client_id'    => $response['clientId'],
+			'redirect_uri' => $response['redirectUri'],
 			'nonce'        => $nonce,
 		];
 	}
+
+	/**
+	 * Perform a remote request for revoking OAuth access for the current user.
+	 *
+	 * @return string The body of the response
+	 * @throws Exception If the remote request fails.
+	 */
+	public function revoke_wpcom_api_auth(): string {
+		$args = [
+			'method'  => 'DELETE',
+			'timeout' => 30,
+			'url'     => $this->get_wpcom_api_url( '/wpcom/v2/sites/' . Jetpack_Options::get_option( 'id' ) . '/wc/partners/google/revoke-token' ),
+			'user_id' => get_current_user_id(),
+		];
+
+		$response = Client::remote_request( $args );
+
+		if ( is_wp_error( $response ) ) {
```

Comment:
Shouldn't we check if the HTTP status code is >= 200? For example, if there are no tokens, the server will respond with status 500 - Internal Server Error. - `{"code":"wpcom_partner_token_not_associated","message":"No token found associated with the client ID and user.","data":null}`

Reply:
Fixed here. Thx https://github.com/woocommerce/google-listings-and-ads/pull/2424/commits/ece73b1dbf2b219230bbac33c5f607dda685a0eb